lib/QubitXmlImport.class.php

```diff
@@ -942,23 +942,25 @@ public static function runFilters(&$node, $filterParam)
   {
     foreach ($filterParam as $filters)
     {
-      foreach ($filters as $eadTag => $object)
+      foreach ($filters as $eadTag => $objects)
       {
-        // Check childnodes for presence of filter node
-        foreach ($node->childNodes as $child)
+        foreach ($objects as $object => $method)
         {
-          // Determine if filtered EAD tag present in child node.
-          if ($child->nodeName == $eadTag)
+          $elementList = $node->getElementsByTagName($eadTag);
+
+          while ($elementList->length > 0)
           {
-            // If so, run specified EAD filter.
-            foreach ($object as $object => $method)
+            $element = $elementList->item(0);
+
+            $parameters = [];
+            if (is_callable(array($object, $method)))
             {
-              if (is_callable(array($object, $method)))
-              {
-                $parameters[] = $eadTag;
-                $parameters[] = $child;
-                $child->nodeValue = call_user_func_array(array( & $object, $method), $parameters);
-              }
+              $parameters[] = $eadTag;
+              $parameters[] = $element;
+              $textValue = call_user_func_array(array( & $object, $method), $parameters);
```

Comment:
You don't need to pass `$object` by reference in this case.

Reply:
I'd also suggest to change `$objects` and `$object` to `$classes` and `$class` in this function to avoid confusion, but that may just be me.

Reply:
I can see the confusion. Will change these names.